src/ConfigFileParser.cpp

```diff
@@ -0,0 +1,534 @@
+}
+
+/**
+ * @brief Processes the line after the directive and value got extracted
+ *
+ * It can be the case that multiple directorives are in the same line.
+ * Therefore the line must still get read until all directives got processed.
+ *
+ * If the line still contains more then one directive, the processed directive gets skipped.
+ * If the line does not contain any more directives, the line gets cleared.
+ *
+ * @param line The current line
+ */
+void ConfigFileParser::processRemainingLine(std::string& line) const
+{
+    size_t semicolonIndex = line.find(';');
+
+    if (line.size() > semicolonIndex + 1) {
+        line = line.substr(semicolonIndex + 1);
+        removeLeadingAndTrailingSpaces(line);
+    } else
+        line.clear();
+}
+
+/**
+ * @brief Reads the current line of the server config and does several checks
+ *
+ * How a line gets processed:
+ * 1. Gets checked if it contains a semicolon.
+ * 2. The directive of of the resulting string gets extracted and checked.
+ * 3. The value of the directive gets extracted and checked.
+ *
+ * If the line contains more then one directive, this process gets repeated for each directive.
+ * Otherwise the line just gets cleared and the processing ends.
+ */
+void ConfigFileParser::readServerConfigLine(void)
+{
+    ConfigServer server;
+    std::string directive;
+    std::string value;
+    std::string line = m_configFile.currentLine;
+
+    initializeConfigServer(server);
+
+    while (!line.empty()) {
+        if (isSemicolonMissing(line))
+            throw std::runtime_error("Semicolon missing");
+
+        directive = getDirective(line);
+        value = getValue(line);
+
+        if ((value.empty() || value.find_last_not_of(" \t\n\v\f\r") == std::string::npos) && directive != "location")
+            throw std::runtime_error("'" + directive + "'" + " directive has no value");
+
+        if (!isDirectiveValid(directive, SERVER))
+            throw std::runtime_error("Invalid server directive");
+        if (directive == "location") {
+            for (readAndTrimLine(); m_configFile.currentLine != "}"; readAndTrimLine())
+                readLocationConfigLine();
+            m_configFile.servers[m_configFile.serverIndex].locationIndex++;
+            break;
+        }
+
+        readServerDirectiveValue(directive, value);
+
+        processRemainingLine(line);
+    }
+}
+
+/**
+* @brief Reads the current line of the location config and does several checks
+*
+* How a line gets processed:
+* 1. Gets checked if it contains a semicolon.
+* 2. The directive of of the resulting string gets extracted and checked.
+* @todo FIXME: 3. The value of the directive gets extracted and checked.
+*
+* If the line contains more then one directive, this process gets repeated for each directive.
+* Otherwise the line just gets cleared and the processing ends.
+
+*/
+void ConfigFileParser::readLocationConfigLine(void)
+{
+    Location location;
+    std::string directive;
+    std::string value;
```

Comment:
Like above: put these in the loop and make em const

Reply:
Done as above.